crates/protect-ffi/src/encrypt_config.rs

```diff
@@ -0,0 +1,490 @@
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct Identifier {
+    #[serde(rename = "t")]
+    pub table: String,
+    #[serde(rename = "c")]
+    pub column: String,
+}
+
+impl Identifier {
+    pub fn new<S>(table: S, column: S) -> Self
+    where
+        S: Into<String>,
+    {
+        let table = table.into();
+        let column = column.into();
+
+        Self { table, column }
+    }
+
+    pub fn table(&self) -> &String {
+        &self.table
+    }
+
+    pub fn column(&self) -> &String {
+        &self.column
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Default)]
+pub struct Tables(HashMap<String, Table>);
+
+impl IntoIterator for Tables {
+    type Item = (String, Table);
+    type IntoIter = std::collections::hash_map::IntoIter<String, Table>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Default)]
+pub struct Table(HashMap<String, Column>);
+
+impl IntoIterator for Table {
+    type Item = (String, Column);
+    type IntoIter = std::collections::hash_map::IntoIter<String, Column>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Default)]
+pub struct EncryptConfig {
+    #[serde(rename = "v")]
+    pub version: u32,
+    pub tables: Tables,
+}
+
+#[derive(Debug, Default, Deserialize, Serialize, Clone, PartialEq)]
+pub struct Column {
+    #[serde(default)]
+    cast_as: CastAs,
+    #[serde(default)]
+    indexes: Indexes,
+}
+
+#[derive(Debug, Default, Clone, Copy, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum CastAs {
+    BigInt,
+    Boolean,
+    Date,
+    Real,
+    Double,
+    Int,
+    SmallInt,
+    #[default]
+    Text,
+    #[serde(rename = "jsonb")]
+    JsonB,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Default, PartialEq)]
+pub struct Indexes {
+    #[serde(rename = "ore")]
+    ore_index: Option<OreIndexOpts>,
+    #[serde(rename = "unique")]
+    unique_index: Option<UniqueIndexOpts>,
+    #[serde(rename = "match")]
+    match_index: Option<MatchIndexOpts>,
+    #[serde(rename = "ste_vec")]
+    ste_vec_index: Option<SteVecIndexOpts>,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
+pub struct OreIndexOpts {}
+
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
+pub struct MatchIndexOpts {
+    #[serde(default = "default_tokenizer")]
+    tokenizer: Tokenizer,
+    #[serde(default)]
+    token_filters: Vec<TokenFilter>,
+    #[serde(default = "default_k")]
+    k: usize,
+    #[serde(default = "default_m")]
+    m: usize,
+    #[serde(default)]
+    include_original: bool,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
+pub struct SteVecIndexOpts {
+    prefix: String,
+}
+
+fn default_tokenizer() -> Tokenizer {
+    Tokenizer::Standard
+}
+
+fn default_k() -> usize {
```

Comment:
Suggest adding `const`.

```rust
const fn default_k()
```

Reply:
Good catch. Protect.js doesn't use the defaults here (it includes them in the Zod schema) so I might remove the defaults on the Rust side to make the behaviour more clear.